sdk/database/helper/credsutil/usernames.go

```diff
@@ -0,0 +1,143 @@
+package credsutil
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+type CaseOp int
+
+const (
+	KeepCase CaseOp = iota
+	Uppercase
+	Lowercase
+)
+
+type usernameBuilder struct {
+	displayName string
+	roleName    string
+	separator   string
+
+	maxLen        int
+	caseOperation CaseOp
+}
+
+func (ub usernameBuilder) makeUsername() (string, error) {
+	userUUID, err := RandomAlphaNumeric(20, false)
+	if err != nil {
+		return "", err
+	}
+
+	now := fmt.Sprint(time.Now().Unix())
+
+	parts := []string{
+		"v",
+		ub.displayName,
+		ub.roleName,
+		userUUID,
+		now,
+	}
+	username := joinNonEmpty(ub.separator, parts...)
+	if ub.maxLen > 0 {
+		username = trunc(username, ub.maxLen)
+	}
+	switch ub.caseOperation {
+	case Lowercase:
+		username = strings.ToLower(username)
+	case Uppercase:
+		username = strings.ToUpper(username)
+	}
+
+	return username, nil
+}
+
+type UsernameOpt func(*usernameBuilder)
+
+func DisplayName(dispName string, maxLength int) UsernameOpt {
+	return func(b *usernameBuilder) {
+		if maxLength > 0 {
```

Comment:
I think it's missing the same where the length is `NoneLength` (-1), which would entirely skip including the display name regardless of what display name was provided. Same for RoleName.

Reply:
Ah good catch. I missed that. Fixed.